### DISCUSSION TITLE
Parameter Models are not correctly displayed in the OpenAPI docs when used both in sub-dependencies and in the endpoint handler

### DISCUSSION THREAD

**Discussion by shaliamekh** [fastapi/fastapi#12252] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Literal, Annotated, Any

from fastapi import Depends, FastAPI, Query

from pydantic import BaseModel, PositiveInt


class PaginationParams(BaseModel):
    page: PositiveInt = 1
    limit: PositiveInt = 30
    sort_by: Literal["created_at", "name"] = "created_at"
    order_by: Literal["desc", "asc"] = "desc"


app = FastAPI()


def pagination_dep(
    pagination_params: Annotated[PaginationParams, Query()]
) -> dict[str, Any]:
    return pagination_params.model_dump()


@app.get("/items")
async def items_collection(
    query_params: Annotated[PaginationParams, Query()],
    pagination_params: Annotated[dict[str, Any], Depends(pagination_dep)],
):
    return {"query_params": query_params.model_dump(), "pagination_params": pagination_params}
```

### Description

Query parameters are duplicated and displayed in a very weird way when I use the same parameter model both in sub-dependency and the endpoint handler as in the example code above.

![image](https://github.com/user-attachments/assets/0188dc3b-dcc6-49f1-8adc-94582d27ac7e)

### Operating System

macOS

### Operating System Details

Sequoia, version 15.0

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

3.12.4

**Comment by YuriiMotov** [comment]
I think this is related to https://github.com/fastapi/fastapi/discussions/12212

To keep the discussions clean, let's close this topic and track it there.
